cmd/snap-confine/snap-confine.c

```diff
@@ -787,20 +764,12 @@ static void enter_non_classic_execution_environment(sc_invocation * inv,
 static void maybe_join_tracking_cgroup(const sc_invocation * inv,
 				       gid_t real_gid, gid_t saved_gid)
 {
-	if (getegid() != 0 && saved_gid == 0) {
-		/* Temporarily raise egid so we can chown the cgroup under LXD. */
-		if (setegid(0) != 0) {
-			die("cannot set effective group id to root");
-		}
-	}
+	/* Temporarily raise egid so we can chown the cgroup under LXD. */
+	sc_identity old = sc_set_effective_identity(sc_root_group_identity());
 	if (!sc_cgroup_is_v2()) {
 		if (sc_feature_enabled(SC_FEATURE_REFRESH_APP_AWARENESS)) {
 			sc_cgroup_pids_join(inv->security_tag, getpid());
 		}
 	}
-	if (geteuid() == 0 && real_gid != 0) {
-		if (setegid(real_gid) != 0) {
-			die("cannot set effective group id to %d", real_gid);
-		}
-	}
+	(void)sc_set_effective_identity(old);
```

Comment:
With the above recommendation of moving this into sc_cgroup_create_and_join() you can just remove the raise drop code here (since sc_cgroup_pids_join() calls it).

Reply:
Done